api/app/auto_spatial_advisory/fuel_type_layer.py

```diff
@@ -1,21 +1,51 @@
-""" Functions for manipulating the BC Fuel Type Layer
-"""
+"""Functions for manipulating the BC Fuel Type Layer"""
+
 import asyncio
-from typing import Generator, Tuple
 import logging
-from osgeo import ogr, osr, gdal
-from shapely import wkt, wkb
+from typing import Generator, Tuple
+
+from osgeo import gdal, ogr, osr
+from shapely import wkb, wkt
+from sqlalchemy.ext.asyncio import AsyncSession
+
 from wps_shared import config
-from wps_shared.db.models.auto_spatial_advisory import FuelType
-from wps_shared.db.database import get_async_write_session_scope
 from wps_shared.db.crud.auto_spatial_advisory import save_fuel_type
+from wps_shared.db.crud.fuel_layer import (
+    get_latest_fuel_type_raster_by_fuel_raster_name,
+    get_processed_fuel_raster_details,
+)
+from wps_shared.db.database import get_async_read_session_scope, get_async_write_session_scope
+from wps_shared.db.models.auto_spatial_advisory import FuelType
 from wps_shared.geospatial.geospatial import NAD83_BC_ALBERS
 from wps_shared.utils.polygonize import polygonize_in_memory
 
-
 logger = logging.getLogger(__name__)
 
 
+async def get_current_fuel_type_raster(session: AsyncSession):
```

Comment:
do we need to pass in a session if we just reopen a read session below? Or perhaps just use the passed in session.